core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala

```diff
@@ -310,6 +334,38 @@ class DumpLogSegmentsTest {
     None
   }
 
+  // Returns the total bytes of the batches specified
+  private def readPartialBatchesBytes(lines: util.ListIterator[String], limit: Int): Int = {
+    val sizePattern: Regex = raw".+?size:\s(\d+).+".r
+    var batchesBytes = 0
+    var batchesCounter = 0
+    while (lines.hasNext) {
+      if (batchesCounter >= limit){
+        return batchesBytes
+      }
+      val line = lines.next()
+      if (line.startsWith("baseOffset")) {
+        line match {
+          case sizePattern(size) => batchesBytes += size.toInt
+          case _ => throw new IllegalStateException(s"Failed to parse and find size value for batch line: $line")
+        }
+        batchesCounter += 1
+      }
+    }
+    return batchesBytes
```

Comment:
No need for return.

Reply:
I agree  in this scenario `limit` always be lower (half) than the total lines, but what if tomorrow we use this method passing a limit which is bigger than the total of batches?  for example the batches = 3 and limit = 5

Reply:
I just meant that in scala you can just do `batchesBytes` instead of `return batchesBytes` in the last statement.